File: aesi-eclipse/plugin/src/main/java/com/virtlink/editorservices/eclipse/DummyProject.java
package com.virtlink.editorservices.eclipse;

import java.net.URI;
import java.net.URISyntaxException;

import com.virtlink.editorservices.IDocument;
import com.virtlink.editorservices.IProject;

public class DummyProject implements IProject {

	private static final long serialVersionUID = -7583455819136225569L;

	@Override
	public URI getUri() {
		try {
			return new URI("dummy");
		} catch (URISyntaxException e) {
			throw new RuntimeException(e);
		}
	}

}

File: aesi-intellij/src/main/kotlin/com/virtlink/editorservices/intellij/DummyProject.kt
package com.virtlink.editorservices.intellij

import java.net.URI

/**
 * A dummy project that is returned when the project
 * a file belongs to cannot be determined.
 */
@Deprecated("Replaced by Project")
object DummyProject {
    override val uri: URI
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
}